src/main/java/edu/ntnu/idi/idatt/boardgame/common/view/MainView.java

```diff
@@ -8,88 +9,87 @@
-    titleLabel.setStyle("-fx-text-fill: white; -fx-font-size: 20px; -fx-font-weight: bold;");
-
-    VBox games = new VBox(10);
-    addGames(games);
-
-    Region spacer = new Region();
-    VBox.setVgrow(spacer, javafx.scene.layout.Priority.ALWAYS);
-
-    Button saveGameButton = new Button("Save game");
-    saveGameButton.setOnAction(e -> System.out.println("Save game not implemented yet."));
-
-    Button loadGameButton = new Button("Load game");
-    loadGameButton.setOnAction(e -> System.out.println("Load game not implemented yet."));
-
-    Button exitButton = new Button("Exit");
-    exitButton.setOnAction(e -> Platform.exit());
-
-    sidebar
-        .getChildren()
-        .addAll(
-            titleLabel,
-            games,
-            spacer,
-            saveGameButton,
-            loadGameButton,
-            exitButton);
-    root.setLeft(sidebar);
-  }
-
-  private void addGames(VBox games) {
-    Button snakesAndLaddersButton = new Button("Snakes and Ladders");
-    snakesAndLaddersButton.setOnAction(e -> loadSnakesAndLadders());
-
-    Button cluedoButton = new Button("Cluedo");
-    cluedoButton.setOnAction(e -> System.out.println("Cluedo not implemented yet."));
-
-    games.getChildren().addAll(snakesAndLaddersButton, cluedoButton);
-  }
-
-  /** Loads the Snakes and Ladders game and sets its view in the center of the root layout. */
-  private void loadSnakesAndLadders() {
-    SnakesAndLaddersController controller = new SnakesAndLaddersController(2);
-    SnakesAndLaddersView view = new SnakesAndLaddersView(controller, controller.getGameBoard());
-    contentWrapper.getChildren().setAll(view.getRoot());
-  }
-}
+    private final BorderPane root;
+    private final StackPane contentWrapper;
+    private GameController currentController;
+    private Button saveGameButton;
+    private Button loadGameButton;
+
+    public MainView() {
+        root = new BorderPane();
+
+        contentWrapper = new StackPane();
+        contentWrapper.setPadding(new Insets(10));
+        root.setCenter(contentWrapper);
+
+        loadMenu();
+    }
+
+    public BorderPane getRoot() {
+        return root;
+    }
+
+    private void loadMenu() {
+        VBox sidebar = new VBox(15);
+        sidebar.setPadding(new Insets(10));
+        sidebar.setAlignment(javafx.geometry.Pos.TOP_CENTER);
+        sidebar.setStyle("-fx-background-color: #336699;");
+
+        Label titleLabel = new Label("Board games");
+        titleLabel.setStyle("-fx-text-fill: white; -fx-font-size: 20px; -fx-font-weight: bold;");
+
+        VBox games = new VBox(10);
+        addGames(games);
+
+        Region spacer = new Region();
+        VBox.setVgrow(spacer, Priority.ALWAYS);
+
+        saveGameButton = new Button("Save game");
+        saveGameButton.setDisable(true);
+        saveGameButton.setOnAction(e -> {
+            if (currentController != null) {
+                currentController.saveGameState("saves/game_state.json");
```

Comment:
We want to be able to save multiple games,  but this always saves the game to the same file.